packages/gatsby-theme-styleguide/src/gatsby-components/Code/Code.js

```diff
@@ -0,0 +1,148 @@
+import React, { useState } from 'react'
+import Highlight, { defaultProps } from 'prism-react-renderer'
+import copyToClipboard from '../../utils/copy-to-clipboard'
+import IconLibrairy from '../IconLibrairy'
+import normalize from './normalize'
+import styled, { css } from 'styled-components'
+
+const CopyButton = styled.button`
+  width: 1.5rem;
+  height: 1.5rem;
+  box-sizing: border-box;
+  cursor: pointer;
+  background: rgb(42, 39, 52);
+  box-shadow: none;
+  border: none;
+  position: absolute;
+  right: 1.1rem;
+  top: 6px;
+  outline: none;
+  z-index: 1000;
+`
+
+const LineNumbered = styled.div`
+  &:before {
+    display: inline-block;
+    width: 3em;
+    overflow: hidden;
+    font-size: 0.8em;
+  }
+`
+
+const Preformated = styled.pre`
+  max-width: 100%;
+  overflow: auto;
+  margin: 0 !important;
+  white-space: pre-wrap;
+  border: none !important;
+  ${({ fullScreen }) =>
+    !fullScreen &&
+    css`
+      max-height: ${({ isOpen }) => (isOpen ? `50vh` : `10vh`)};
+    `}
+
+  ${({ fullScreen }) =>
+    fullScreen &&
+    css`
+      width: 100%;
+      max-height: none;
+      overflow: none;
+
+      code {
+        overflow: none;
+      }
+    `};
+`
+
+const Body = styled.div`
+  flex: 1;
+  overflow: auto;
+  position: relative;
+  border-left: solid 1px #ececec;
+`
+
+const copyToClipboardClick = (str, toogleCopy) => {
+  toogleCopy(true)
```

Comment:
Try again one last time ;)

You want to toggle the state only if the copy worked, if not, return null or false. 
No console logs in front-end code. 

if there is an error, your state will stay `copied: true` which is not what we want.

Reply:
👍  Got it! Almost there :)